### PR TITLE
Pub enum runtime to pub struct runtime

### PR DESCRIPTION
**Pull request by a-moreira**
This PR simply substitutes `Pub enum Runtime` for `Pub struct Runtime` in the `construct_runtime!` macros, as suggested by @kianenigma on the Polkadot Blockchain Academy this week :-)

polkadot address: 14WCnNFaTCYW2BTS2rDMHacvWdfQEPp2kJjAsp9VdHJJrNNT

### CODE REVIEW

**Comment by ggwpez** [comment]
This would be a breaking change for all downstream projects…  
@kianenigma sure that this is worth it?

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -1701,7 +1701,7 @@ impl frame_benchmarking_pallet_pov::Config for Runtime {
 }
 
 construct_runtime!(
-	pub enum Runtime where
+	pub struct Runtime where
```
Comment (ggwpez) [inline bin/node/runtime/src/lib.rs:1704]: Did you change the `construct_runtime` macro itself?  
Otherwise this obviously wont compile.

Comment (shawntabrizi) [reply bin/node/runtime/src/lib.rs:1704]: Already done: https://github.com/paritytech/substrate/pull/11932

**Comment by shawntabrizi** [comment]
This should be non-breaking and work out of the box afaik.

So yeah, I think its worth.

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply bin/node/runtime/src/lib.rs:1704]: At last academy 😜

Comment (ggwpez) [reply bin/node/runtime/src/lib.rs:1704]: Ah okay. The failing UI tests were from a change to their files, I see.

**Comment by kianenigma** [comment]
This should not be breaking as Shawn said. Just use `pub struct Runtime` in our codebase, and mark usage of `pub enum Runtime` as deprecated.

**Review by kianenigma — APPROVED** [review]
I'm down to mark the usage of `pub enum whatever` as deprecated as well, and then remove it. 

If so others agree, please do that in this PR as well.

**Comment by xlc** [comment]
What's the reason for doing this?

File: frame/support/test/tests/construct_runtime.rs [hunk 2]
```diff
@@ -254,7 +254,7 @@ impl system::Config for Runtime {
 }
 
 frame_support::construct_runtime!(
-	pub enum Runtime where
+	pub struct Runtime where
```
Comment (xlc) [inline frame/support/test/tests/construct_runtime.rs:257]: `pub enum Runtime` is still supported so we should keep the old test and add new one for `pub struct Runtime`

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2335074

**Comment by shawntabrizi** [comment]
> What's the reason for doing this?

We are at the academy, and it is just more clear to use a `struct` than an `enum` for the "runtime" container.

**Comment by kianenigma** [comment]
bot rebase

**Comment by kianenigma** [comment]
CI needs work @a-moreira

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2413644

**Comment by a-moreira** [comment]
@kianenigma sorry for the delay, I was AFK in the last days. Should be fixed now

**Comment by a-moreira** [comment]
@kianenigma CI works. I updated `trybuild` error files (like this approach https://github.com/paritytech/substrate/pull/8372) and I created another file to keep a test for `Pub enum Runtime`. no asnwers, so assuming it would be better to deprecate the other stuff in another PR.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot rebase

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-40/2468/1

**Pull request merged**